portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+
+import com.google.appengine.api.datastore.FetchOptions;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/delete-data")
+public class DeleteServlet extends HttpServlet {
+
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Query query = new Query("Comment");
+    PreparedQuery results = datastore.prepare(query);
+
+    for(Entity entity : results.asIterable()){
+        datastore.delete(entity.getKey());
+    }
```

Comment:
This could be a good candidate for Streams. I know we haven't talked much about streams, but they are pretty awesome and we can talk some more. Streams also touch the topic of lambdas which we can discuss in a vc.

I've left some more details below if you're interested, but feel free to go with the for-loop till we discuss and build more comfort with streams.

Often streams are extremely efficient, easy to use, and compact. A gauge of when to use them is usually how easily you can go from the for-loop version of the code to a stream analog. If it's requiring a lot of work, streams might not be fit.

So something like this might work well:
`results.asIterable().stream().forEach(entity -> datastore.delete(entity.getKey()))`

Reply:
I actually attempted to use streams with this and got stuck. I tried to map the Entity stream to Keys then call a forEach loop on them that deletes the keys from the datastore, but I kept getting an error. This makes sense to just call the getKey() method on each Entity.

Reply:
I was actually trying to use streams for this but it wasn't working because .asItrerable() returns an Iterable object. I can't call .stream() on it because the method isn't in the interface. The same thing happened with .asIterator() too.

Reply:
Oh nevermind, I found a solution using Spliterators.